sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -963,6 +964,53 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
     )
   }
 
+  test("SPARK-13764 Parse modes in JSON data source") {
```

Comment:
split it into 2 cases: `Corrupt records: FAILFAST mode` and `Corrupt records: DROPMALFORMED mode`